rotkehlchen/api/v1/schemas.py

```diff
@@ -734,49 +739,155 @@ def generate_fields_post_validation(self, data: dict[str, Any]) -> dict[str, Any
+            ),
+        )
+        validator_index = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Validator index must be an integer >= 0',
+            ),
+        )
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            return {'event': EthBlockEvent(**data)}
+
+    class CreateEthDepositEventEventSchema(BaseSchema):
+        tx_hash = EVMTransactionHashField(required=True)
+        depositor = EvmAddressField(required=True)
+        event_identifier = fields.String(required=True)
+        sequence_index = fields.Integer(required=True)
+        validator_index = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Validator index must be an integer >= 0',
+            ),
+        )
+        extra_data = fields.Dict(load_default=None)
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            return {'event': EthDepositEvent(**data)}
+
+    class CreateEthWithdrawalEventEventSchema(BaseSchema):
+        is_exit = fields.Boolean(required=True)
+        withdrawal_address = EvmAddressField(required=True)
+        validator_index = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Validator index must be an integer >= 0',
+            ),
+        )
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            return {'event': EthWithdrawalEvent(**data)}
+
+    ENTRY_TO_SCHEMA: Final[dict[HistoryBaseEntryType, type[Schema]]] = {
+        HistoryBaseEntryType.HISTORY_EVENT: CreateBaseHistoryEventSchema,
+        HistoryBaseEntryType.ETH_BLOCK_EVENT: CreateEthBlockEventEventSchema,
+        HistoryBaseEntryType.ETH_DEPOSIT_EVENT: CreateEthDepositEventEventSchema,
+        HistoryBaseEntryType.ETH_WITHDRAWAL_EVENT: CreateEthWithdrawalEventEventSchema,
+        HistoryBaseEntryType.EVM_EVENT: CreateEvmEventSchema,
+    }
 
     @post_load
     def make_history_base_entry(
             self,
             data: dict[str, Any],
             **_kwargs: Any,
     ) -> dict[str, Any]:
-        if data.get('event_subtype', None) is None:
-            data['event_subtype'] = HistoryEventSubType.NONE
-
-        if data['location'] not in EVM_LOCATIONS:
-            raise ValidationError(
-                message=f'EVM event location needs to be one of {EVM_LOCATIONS}',
-                field_name='location',
-            )
+        entry_type = data.pop('entry_type')
```

Comment:
Just explain why you pop
```suggestion
        entry_type = data.pop('entry_type')  # already used to decide schema
```